src/base.service.ts

```diff
@@ -206,8 +206,35 @@ export abstract class BaseService<Entity extends object, CreateModel, UpdateMode
     filters?: unknown,
     info?: Info
   ): SelectQueryBuilder<Entity> {
+    return this._processBaseFilters(query, filters, Object.keys(filters || {}));
+  }
+
+  private _processBaseFilters<Filters>(
+    query: SelectQueryBuilder<Entity>,
+    filters: Filters,
```

Comment:
This could be narrowed to `Filters extends Record<string, unknown>` or even `Filters extends Record<string, string[]>`, but then this would require adding the same generic to `setFilters()`, or make `Filters` a class-level generic. I'd be in favour of leaving this as is in the current PR for the moment.